test/telemetry/agent/test_client_server.rb

```diff
@@ -182,20 +182,21 @@ def wait_locked(timeout)
                 end
 
                 DataChannel = Struct.new :op, :queue, :thread, keyword_init: true do # rubocop:disable Metrics/BlockLength
-                    def self.setup(client)
+                    def self.setup(client, timeout: 5)
                         data_op = client.data(Grpc::Void.new, return_op: true)
                         sample_queue = QueueWithTimeout.new
 
                         channel_ready = Concurrent::Event.new
                         pull_thread = Thread.new do
                             enum = data_op.execute
-                            channel_ready.set
                             enum.each do |sample|
-                                sample_queue.enq(sample)
+                                # We ignore the first sample. It is sent as a way to
+                                # confirm that the channel is successfully established
+                                sample_queue.enq(sample) unless channel_ready.set?
                             end
                         rescue GRPC::Core::CallError, GRPC::Cancelled # rubocop:disable Lint/SuppressedException
                         end
-                        channel_ready.wait
+                        channel_ready.wait(timeout)
```

Comment:
Where is channel_ready being set?

Reply:
Good catch.

The set? above was meant to be a try? (which sets only if not yet set, and returns true in that case). And `Event#wait(timeout)` returns a boolean. I was assuming it to raise...

Both fixed.